sklearn/preprocessing/imputation.py

```diff
@@ -374,3 +380,487 @@ def transform(self, X):
+            "n_features" is the number of features.
+
+        Returns
+        -------
+        X : array-like, shape (n_samples, n_features)
+            The imputed input data.
+        """
+        if self.estimator is None:
+            from ..linear_model import BayesianRidge
+            self.estimator_ = BayesianRidge()
+        else:
+            self.estimator_ = clone(self.estimator)
+
+        # check that the estimator's predict method has return_std argument
+        if 'return_std' not in signature(self.estimator_.predict).parameters:
+            raise ValueError(
+                "The regression estimator is %s and its predict method does "
+                "not support 'return_std=True'. This is required for MICE." %
+                type(self.estimator_)
+            )
+
+        # pre-processing of X
+        X = check_array(X, dtype=np.float32, order="F", force_all_finite=False)
+
+        # parse min and max values
+        self.min_value_ = np.nan if self.min_value is None else self.min_value
+        self.max_value_ = np.nan if self.max_value is None else self.max_value
+
+        # initial imputation
+        mask_missing_values = _get_mask(X, self.missing_values)
+        self.initial_imputer_ = Imputer(missing_values=self.missing_values,
+                                        strategy=self.initial_strategy,
+                                        axis=0)
+        X_filled = self.initial_imputer_.fit_transform(X)
+
+        valid_mask = np.logical_not(np.isnan(
+            self.initial_imputer_.statistics_
+        ))
+        self._valid_statistics_indexes = np.flatnonzero(valid_mask)
+        X = X[:, self._valid_statistics_indexes]
+        mask_missing_values = mask_missing_values[
+                              :,
+                              self._valid_statistics_indexes
+        ]
+
+        # perform imputations
+        self._trained_estimator_triplets = []
+        n_samples, n_features = X_filled.shape
+        total_rounds = self.n_burn_in + self.n_imputations
+        results_list = []
+        if self.verbose:
+            print("[MICE] Completing matrix with shape %s" % (X.shape,))
+            start_t = time()
+            mice_msg = '[MICE] Ending imputation round '
+        for iter in range(total_rounds):
+            # order in which to impute
+            ordered_idx = self._get_ordered_idx(mask_missing_values)
+
+            # abs_correlation matrix is used to choose a subset of other
+            # features to impute from
+            abs_corr_mat = self._get_abs_correlation_matrix(X_filled)
+
+            # Fill in each feature in the order of ordered_idx
+            for feat_idx in ordered_idx:
+                neighbor_feat_idx = self._get_neighbor_feat_idx(
+                    n_features,
+                    feat_idx,
+                    abs_corr_mat
+                )
+                X_filled, estimator = self._impute_one_feature(
+                    X_filled,
```

Comment:
don't use multiple lines

Reply:
If I do this, it goes past 80 chars:
```
                X_filled, estimator = self._impute_one_feature(X_filled,
                                                               mask_missing_values,
                                                               feat_idx,
                                                               neighbor_feat_idx)
```

What's another option?